src/components/FilterAnnotMondal/FilterAnnotModal.js

```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import defaultTool from 'constants/defaultTool';
+import core from 'core';
+import actions from 'actions';
+import selectors from 'selectors';
+
+import Choice from 'components/Choice';
+import Button from 'components/Button';
+
+import { Swipeable } from 'react-swipeable';
+import { FocusTrap } from '@pdftron/webviewer-react-toolkit';
+
+import './FilterAnnotModal.scss';
+
+const FilterAnnotModal = () => {
+  const [isDisabled, isOpen] = useSelector(state => [
+    selectors.isElementDisabled(state, 'filterModal'),
+    selectors.isElementOpen(state, 'filterModal'),
+  ]);
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+
+  const [authors, setAuthors] = useState([]);
+  const [annotTypes, setAnnotTypes] = useState([]);
+
+  const [authorFilter, setAuthorFilter] = useState([]);
+  const [typesFilter, setTypesFilter] = useState([]);
+
+  const filterApply = () => {
+    dispatch(
+      actions.setCustomNoteFilter(annot => {
+        let type = true;
+        let author = true;
+        if (typesFilter.length > 0) {
+          type = typesFilter.includes(annot.Subject.toLowerCase().replace(/\s/g, ''));
+        }
+        if (authorFilter.length > 0) {
+          author = authorFilter.includes(annot.Author);
+        }
+        return type && author;
+      }),
+    );
+    closeModal();
+  };
+
+  const filterClear = () => {
+    dispatch(
+      actions.setCustomNoteFilter(annot => {
+        return true;
+      }),
+    );
+    setAuthorFilter([]);
+    setTypesFilter([]);
+    closeModal();
+  };
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('filterModal'));
+    core.setToolMode(defaultTool);
+  };
+
+  useEffect(() => {
+    const annots = core.getAnnotationsList();
+    // set is a great way to remove any duplicate additions and ensure the unique items are present
+    // the only gotcha that it should not be used by state since not always it will trigger a rerender
+    const authorsToBeAdded = new Set();
+    const annotTypesToBeAdded = new Set();
+    annots.forEach(annot => {
+      if (core.getDisplayAuthor(annot) && core.getDisplayAuthor(annot) !== '') {
+        authorsToBeAdded.add(core.getDisplayAuthor(annot));
+      }
+      if (annot.Subject && annot.Subject !== '') {
+        // we want to lowercase it and remove spaces to get i18n translations
```

Comment:
Thanks for putting the comment. Now I'm wondering if we need to display the translated text?

For annotations created inside WebViewer, their `Subject` have already been translated using i18n. See the code at `setupI18n.js`.
```js
window.Annotations.Utilities.setAnnotationSubjectHandler(type =>
  t(`annotation.${type}`),
);
```
Since it's already translated I feel like it's not necessary to do a `lowerCase` then `replace` here and then render `t(annotation.${val})`. `annotation.Subject` here is already `t(annotation.${val})`.

For annotations created by users in other PDF viewers, we don't have existing translations for their `Subject`s, so we don't need to do these string manipulations.